kore/src/Kore/Step/Simplification/Pattern.hs

```diff
@@ -59,19 +63,31 @@ simplifyTopConfiguration ::
 simplifyTopConfiguration =
     simplify >=> return . removeTopExists
 
-{- | Simplifies the 'Pattern' with the assumption that the 'TermLike' is defined
+{- | Simplifies the 'Pattern', with the assumption that the term is defined,
 and removes the exists quantifiers at the top.
 -}
 simplifyTopConfigurationDefined ::
     MonadSimplify simplifier =>
     Pattern RewritingVariableName ->
-    TermLike RewritingVariableName ->
     simplifier (OrPattern RewritingVariableName)
-simplifyTopConfigurationDefined patt defined =
-    makeEvaluate sideCondition patt
-        >>= return . removeTopExists
+simplifyTopConfigurationDefined configuration =
+    maybe
+        (return OrPattern.bottom)
+        (worker definedConfiguration)
+        sideCondition
   where
-    sideCondition = SideCondition.assumeDefined defined
+    worker patt condition =
+        makeEvaluate condition patt
+            >>= return . removeTopExists
+
+    term = Conditional.term configuration
+    sideCondition = SideCondition.assumeDefined term
+    definedConfiguration =
+        Pattern.andCondition
+            configuration
+            ( makeCeilPredicate term
+                & from @_ @(Condition _)
+            )
```

Comment:
I think this reads a little better, because the most important bits are first and last:

```suggestion
        makeCeilPredicate term
            & from @_ @(Condition _)
            & Pattern.andCondition configuration
```

The important bits are: we are adding a `\ceil`, and we are adding it to the condition.